x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java

```diff
@@ -103,6 +103,20 @@ protected void masterOperation(
         createFollowerIndexAndFollowRemoteIndex(request, leaderCluster, leaderIndex, listener);
     }
 
+    /**
+     * Pre-flight check if the given index can be a leader index.
+     * This method returns {@code null} if no issue found; otherwise it returns the first issue that it found.
+     */
+    static Exception canFollow(String leaderCluster, IndexMetaData leaderIndexMetaData) {
```

Comment:
I don't think this needs to be a method shared between `AutoFollowCoordinator` and `TransportPutFollowAction`; I think that we can inline the check directly into the call sites. Then we can avoid having a debate about the method not throwing versus returning null versus returning an optional. 😇

Reply:
If we think there is value in having this method (for example, maybe for future purposes), then let us discuss that.

Reply:
The reason I introduced this method is for future purposes when we need to add one more check for the leader index. I am fine to inline it.

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/34767/commits/b279bd78bfb2e16c79b4607786440c7126e0cf14.